src/graphql/schema/User.ts

```diff
@@ -41,11 +49,82 @@ export const UserQuery = extendType({
 			type: "User",
 		});
 		t.list.field("users", {
+			authorize: (_root, _args, ctx) => {
+				const map = ctx.req.session.user!.permissions!.map(
+					(permOnUsers) => permOnUsers.permission.name,
+				);
+				// only if a user can mutate all users he should be able
+				// to query all users
+				return map.includes("canMutateUsers");
+			},
```

Comment:
This can be turned into a one-liner

Reply:
oh yeah right. It originally was but i wanted to log the map variable.